File: softbar/src/app/services/domicilio/domicilio.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Domicilio } from 'src/app/models/Domicilio/domicilio';
import { Pedido } from 'src/app/models/Domicilio/pedido';
import baserUrl from '../helper';

@Injectable({
  providedIn: 'root'
})
export class DomicilioService {

  url: String = 'http://localhost:8083/domicilio';


  constructor(private http: HttpClient) { }

  getDomicilio(): Observable<Domicilio[]>{
    return this.http.get<Domicilio[]>(this.url + '/listar');
  }

  getPedido(): Observable<Pedido[]> {
    return this.http.get<Pedido[]>(this.url + '/pedido')
  }

  public CancelarDomicilio(domicilio ){
    return this.http.post(`${baserUrl}/domicilio/cancelado`,domicilio);
  }

  crearDomicilio(domicilio: Domicilio): Observable<Domicilio> {
    return this.http.post<Domicilio>(this.url + '/agregar', domicilio);
  }

  obtenerDomicilio(id_domicilio: number): Observable<Domicilio> {
    return this.http.get<Domicilio>(this.url + '/' + id_domicilio);
  }

  actualizarDomicilio(domicilio: Domicilio): Observable<Domicilio> {
    return this.http.put<Domicilio>(this.url + '/actualizar/' + domicilio.id_domicilio, domicilio)
  }

  eliminarDomicilio(id_domicilio: number): Observable<any> {
    return this.http.delete<any>(this.url + '/eliminar/' + id_domicilio);
  }

}

File: Softbarv4/src/main/java/com/Softbarv4/Softbarv4/IService/domicilioService.java
package com.Softbarv4.Softbarv4.IService;

import java.util.List;

import com.Softbarv4.Softbarv4.Model.Domicilio;

public interface domicilioService {
	
	public List<Domicilio> listarDomicilio();
	
	public Domicilio guardarDomicilio(Domicilio domicilio);
	
	public Domicilio domicilioPorId(Long id_domicilio);
	
	public Domicilio actualizarDomicilio(Domicilio domicilio);
	
	public Domicilio cancelarDomicilio(Domicilio domicilio);

	public void eliminarDomicilio(Long id_domicilio);
	


}
